src/main/java/seedu/address/model/person/Room.java

```diff
@@ -21,15 +22,33 @@
      */
     public Room(String room) {
         requireNonNull(room);
-        checkArgument(isValidRoom(room), MESSAGE_CONSTRAINTS);
+        checkArgument(isValidRoom(room), getMessageConstraints());
         value = room;
     }
 
     /**
      * Returns true if a given string is a valid Room number.
      */
     public static boolean isValidRoom(String test) {
-        return test.matches(VALIDATION_REGEX);
+        if (!test.matches(VALIDATION_REGEX)) {
+            return false;
+        }
+        int room = Integer.parseInt(test);
+        boolean isRoomValid = room % 100 <= roomPref % 100 && room % 100 > 0;
+        boolean isFloorValid = Math.floorDiv(room, 100) <= Math.floorDiv(roomPref, 100)
+                && Math.floorDiv(room, 100) > 0;
```

Comment:
Perhaps can declare MIN, MAX_FLOOR_NUMBER and MIN, MAX_ROOM_NUMBER as a static variable and use it to validate room and floor? It may be clearer and more readable.